### ISSUE TITLE
Sharing permission issue

### ISSUE THREAD

**Issue by MorrisJobke** [owncloud/core#22739]
### Steps to reproduce
1. user0, user1, user2 exist
2. user0 shares folder A to user1(read-write) and user2 (read-only)
3. user1 logs in and open the sidebar for received share A
4. user0 removes the "can edit" permission for user1
5. user1 (didn't refresh the browser) could still give user2 the "can edit" permission

note: after a refresh user1 couldn't see the "can edit" permission - so this seems to be possible from the API only and seems to be not properly checked.

### Expected behavior
* user1 could not do 5.

### Actual behavior
* user1 could do 5.

current master

Quite critical - cc @rullzer @PVince81 @LukasReschke

**Comment by MorrisJobke** [comment]
@cmonteroluque @PVince81 I put this into 9.0.1, but given that this is kind of critical I would be more than happy to have this solved in 9.0.

**Comment by MorrisJobke** [comment]
Initial share setup (read mjob as user0) - after step 3:

![bildschirmfoto 2016-02-29 um 18 27 10](https://cloud.githubusercontent.com/assets/245432/13402906/c3356c94-df12-11e5-8b5c-b1a5599061ee.png)

**Comment by rullzer** [comment]
Duplicate of https://github.com/owncloud/core/issues/22675
